builtin/providers/aws/resource_aws_s3_bucket.go

```diff
@@ -980,7 +980,7 @@ func resourceAwsS3BucketDelete(d *schema.ResourceData, meta interface{}) error {
 				return resourceAwsS3BucketDelete(d, meta)
 			}
 		}
-		return fmt.Errorf("Error deleting S3 Bucket: %s", err)
+		return fmt.Errorf("Error deleting S3 Bucket: %s '%s'", err, d.Get("bucket"))
```

Comment:
this needs to be d.Get("bucket").(string) - it needs to have the cast

Also, instead of '%s' you can use %q